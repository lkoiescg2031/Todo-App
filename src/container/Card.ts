import { connect } from "react-redux";

import Card from "../components/Card";

import {
	updateItem as requestUpdateCard,
	deleteItem as requestDeleteCard,
} from "../redux/actions/cardAction";

const mapStateToProps = (state: {}) => {
	return {};
};

const mapDispatchToProps = {
	requestUpdateCard,
	requestDeleteCard,
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
