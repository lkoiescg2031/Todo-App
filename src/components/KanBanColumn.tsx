import React, { MouseEvent, PureComponent } from "react";

import { withStyles } from "@material-ui/core/styles";

import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";

import grey from "@material-ui/core/colors/grey";

import CardCreateButton from "./CardCreateButton";
import ColumnMenuButton from "./ColumnMenuButton";
import ColumnTitle from "./ColumnTitle";

interface KanBanColumnProp {
	name: string;
	classes: {
		columns: string;
		works: string;
	};
}

interface KanBanColumnState {
	isTitleEdit: boolean;
}

export const Width = "260px";

class KanBanColumns extends PureComponent<KanBanColumnProp, KanBanColumnState> {
	constructor(props: KanBanColumnProp) {
		super(props);

		this.state = {
			isTitleEdit: false,
		};
		this.onColumnNameClicked = this.onColumnNameClicked.bind(this);
	}

	onColumnNameClicked(e: MouseEvent) {
		this.setState((prevState) => ({
			...prevState,
			isTitleEdit: !prevState.isTitleEdit,
		}));
	}

	render() {
		const { classes, children, name } = this.props;
		const childNode = (children as Array<any>).filter(
			(child) => typeof child !== "boolean"
		)[0];

		const appendCardText =
			Array.isArray(childNode) && childNode.length === 0
				? "Add a card"
				: "Add another card";

		return (
			<Card elevation={2} classes={{ root: classes.columns }}>
				<CardHeader
					title={<ColumnTitle value={name} initialMode="text" />}
					action={<ColumnMenuButton />}
				/>
				<div className={classes.works}>
					{children}
					<CardCreateButton text={appendCardText} />
				</div>
			</Card>
		);
	}
}

export default withStyles((theme) => ({
	columns: {
		minWidth: Width,
		height: "max-content",
		margin: `0px ${theme.spacing(1)}px`,
		padding: theme.spacing(1),
		background: grey[100],
	},
	works: {
		"& > div": {
			margin: `${theme.spacing(1)}px 0px`,
		},
	},
}))(KanBanColumns);
