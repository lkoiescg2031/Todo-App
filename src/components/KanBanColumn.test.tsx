import React from "react";
import { render } from "@testing-library/react";

import KanBanColumn from "./KanBanColumn";
import Card from "./Card";

describe("<KanBanColumn />", () => {
	it("snapshot test", () => {
		const columnName = "TODO";
		const cardNames = ["Work 1", "Work 2", "Work 3"];
		const utils = render(
			<KanBanColumn name={columnName}>
				{cardNames.map((name, idx) => (
					<Card key={`work-${idx}`} title={name} />
				))}
			</KanBanColumn>
		);

		expect(utils.container).toMatchSnapshot();
	});

	it("column name test", () => {
		const columnName = "TODO";
		const utils = render(<KanBanColumn name={columnName} />);

		const span = utils.getByText(columnName);
	});

	it("no child append name test", () => {
		const appendButtonName = "Add a card";
		const utils = render(<KanBanColumn name="Test" />);

		const button = utils.getByText(appendButtonName);
	});

	it("has children append name test", () => {
		const appendButtonName = "Add another card";
		const utils = render(
			<KanBanColumn name="Test">
				<Card title="Work 1" />
			</KanBanColumn>
		);

		utils.getByText(appendButtonName);
	});

	it("children render test", () => {
		const childrenNames = ["Work1", "Work2", "Work3"];
		const utils = render(
			<KanBanColumn name="Test">
				{childrenNames.map((name, idx) => (
					<KanBanColumn key={`card=${idx}`} name={name} />
				))}
			</KanBanColumn>
		);

		childrenNames.forEach((name) => {
			utils.getByText(name);
		});
	});
});
