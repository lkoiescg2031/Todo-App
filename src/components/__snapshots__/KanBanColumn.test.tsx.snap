// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KanBanColumn /> snapshot test 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root KanBanColumns-columns-1 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          <span
            class="EditableText-spanText-4"
          >
            TODO
          </span>
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="KanBanColumns-works-2"
    >
      <div
        class="MuiPaper-root MuiCard-root KanBanCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root KanBanCard-actionArea-6"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Work 1
          </p>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiCard-root KanBanCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root KanBanCard-actionArea-6"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Work 2
          </p>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiCard-root KanBanCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root KanBanCard-actionArea-6"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Work 3
          </p>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root KanBanColumns-addButton-3 MuiButton-text MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Add another card
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
